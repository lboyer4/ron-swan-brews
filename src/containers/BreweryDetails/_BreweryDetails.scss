.big-card {
	height: 300px;
	width: 300px;
	margin: auto;
	border: 2px solid #A57F60;
  background-color: #F7DBA7;
  opacity: .9;
}

.background-big {
	background-image: url(https://www.geek.com/wp-content/uploads/2017/02/ron-swanson-625x352.jpg)
}
