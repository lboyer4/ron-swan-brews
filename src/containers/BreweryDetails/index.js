import React, { Component } from 'react';
import './_BreweryDetails.scss';
import { connect } from 'react-redux';
import { toggleFavorite } from '../../actions';

export class BreweryDetails extends Component {
	constructor() {
		super()
		this.state = {

		}
	}

	handleFavorite = () => {
		this.props.toggleFavorite(this.props.id)
	}

	render() {
		const { name, brewery_type, city, street, phone } = this.props
		
		return (
			<section>
			<div className="big-card">
				<h2>{name}</h2>
				<p className="brew type">{brewery_type}</p>
				<h6 className="brew city">{city}</h6>
				<h6 className="brew street">{street}</h6>
				<h6 className="brew number">{phone}</h6>
				<button onClick={this.handleFavorite}>Favorite</button>
			</div>
			</section>
		)
	}
}

export const mapDispatchToProps = (dispatch) => ({
	toggleFavorite: (id) => dispatch(toggleFavorite(id))
})

export default connect(null, mapDispatchToProps)(BreweryDetails);

