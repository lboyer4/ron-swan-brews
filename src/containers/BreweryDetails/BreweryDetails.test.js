import React from 'react';
import { shallow } from 'enzyme';
import { BreweryDetails } from '../BreweryDetails';

describe('BreweryDetails', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(
			<BreweryDetails
			/>
		);
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	});
});