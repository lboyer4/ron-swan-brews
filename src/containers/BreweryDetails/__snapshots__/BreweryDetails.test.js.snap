// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryDetails should match snapshot 1`] = `
<section>
  <div
    className="big-card"
  >
    <h1 />
    <h4
      className="brew type"
    >
      Brewery Type: 
    </h4>
    <h4
      className="brew city"
    >
      City: 
    </h4>
    <h4
      className="brew street"
    >
      Street: 
    </h4>
    <h4
      className="brew number"
    >
      Phone: 
    </h4>
    <button
      className="favorite-btn"
      onClick={[Function]}
    >
      Favorite
    </button>
  </div>
</section>
`;
